merqube_client_lib/secapi/client.py

```diff
@@ -1,321 +1,9 @@
-        }
-
-        if metrics_chunk_size is None and securities_chunk_size is None:
-            # no chunking
-            data = self._get_security_metrics_helper(**params)  # type: ignore
-            return pd.json_normalize(data, max_level=normalize_level)
-
-        self._validate_chunking_options(
-            addl_options=addl_options,
-            metrics=metrics,
-            sec_names=sec_names,
-            sec_ids=sec_ids,
-            metrics_chunk_size=metrics_chunk_size,
-            securities_chunk_size=securities_chunk_size,
-        )
-
-        dfs: list[pd.DataFrame] = []
-
-        if metrics_chunk_size is not None:
-            for chunk in batch_post_payload(rows=list(metrics), batch_size=metrics_chunk_size):
-                params["metrics"] = chunk
-                data = self._get_security_metrics_helper(**params)  # type: ignore
-                df = pd.json_normalize(data, max_level=normalize_level)
-
-                # when we chunk by metrics, we may be missing some becuase the secapi doesnt return it if its None for all records
-                for m in metrics:
-                    if m not in df:
-                        df[m] = None
-                dfs.append(df)
-
-        elif securities_chunk_size is not None:
-            if sec_names:
-                for chunk in batch_post_payload(rows=list(sec_names), batch_size=securities_chunk_size):
-                    params["sec_names"] = chunk
-                    data = self._get_security_metrics_helper(**params)  # type: ignore
-                    dfs.append(pd.json_normalize(data, max_level=normalize_level))
-            elif sec_ids:
-                for chunk in batch_post_payload(rows=list(sec_ids), batch_size=securities_chunk_size):
-                    params["sec_ids"] = chunk
-                    data = self._get_security_metrics_helper(**params)  # type: ignore
-                    dfs.append(pd.json_normalize(data, max_level=normalize_level))
-
-        # the groupbys below squashes
-        # eff1 id1 m1=NAN m2=x
-        # eff2 id1 m1=Y  m2=NAN
-        # into
-        # eff1 id1 m1=Y m2=x
-        # also:
-        # for chunked, we return a consistent sort order of id, eff_ts
-        return (
-            pd.concat(dfs)
-            .groupby(["eff_ts", "id"])
-            .last()
-            .reset_index()
-            .sort_values(["id", "eff_ts"])
-            .reset_index()
-            .drop("index", axis=1)
-        )
-
-
-secapi_client_cache: LRUCache = LRUCache(maxsize=256)  # type: ignore
-
-
-@cached(cache=secapi_client_cache)
-def get_client(
-    user_session: Optional[MerqubeAPISession] = None, token: Optional[str] = None, **session_kwargs: Any
-) -> SecAPIClient:
-    """
-    Cached; returns a secapi client for token
-    """
-    return SecAPIClient(user_session=user_session, token=token, **session_kwargs)
+SecAPIClient = _SecAPIClient
+secapi_client_cache = merqube_client.client_cache
+get_client = merqube_client.get_client
```

Comment:
backwards compatible references

Reply:
this get_client is returning a MerQubeAPIClient now instead of a SecAPI Client. It should have the expected methods, but will this break typing in places where we do

```
client: SecAPIClient = secape.client.get_client()
```

I don't know the precise behaviour when you have multiple inheritance.

Reply:
Alright - ill repeat the method to return the type here

Reply:
actually, maybe ill just leave this module where it was, but inherit from it